### PR TITLE
[#1] As a developer, I can create a mock project based on a template

### LINKED ISSUE #1: As a developer, I can create a mock project based on a template.
## Why

We need to support creating a mock server with [Prism](https://github.com/stoplightio/prism).

## Acceptance Criteria

- Have OpenAPI specification
- Have Prism create a mock server with Docker support
- Have GitHub Actions CD workflows to support automatically deploy to fly.io

## Resources

- Example: https://github.com/nimblehq/nimble-survey-web/pull/104
- Instructions: https://www.notion.so/nimblehq/API-Mock-Server-with-Prism-396c1c62064c460586958535846f0dc2?pvs=4

### PR DESCRIPTION
**Pull request by suho**
- Close #1

## What happened 👀

Init a project for mock templates:
  - OpenAPI spec with `/health` endpoint as an example.
  - CD workflows for deploying fly.io and lint.
  - 

## Insight 📝

- I created a project to support creating a mock server based on OpenAPI specs.
- Most of the works are based on the https://github.com/nimblehq/nimble-survey-web project.
- Developers can create new mock servers quickly with this templates.

## Proof Of Work 📹

- Mock Server for the Survey project: https://github.com/nimblehq/nimble-survey-web/pull/104.
- Lint CI passed.

### CODE REVIEW

File: Dockerfile.mock
Comment (hoangmirs) [inline Dockerfile.mock:1]: Should we rename this file into `Dockerfile` without `.mock`? That is the main Dockerfile, so I think omitting the `.mock` will be better

File: fly.toml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+app = "api-mock-server"
+primary_region = "sin"
+
+[build]
+  dockerfile = "Dockerfile.mock"
```
Comment (hoangmirs) [inline fly.toml]: Regarding previous comment, we should change here as well:
```suggestion
  dockerfile = "Dockerfile"
```

File: Dockerfile.mock
Comment (suho) [reply Dockerfile.mock:1]: Fixed in https://github.com/nimblehq/api-mock-templates/pull/3/commits/9cb6f6d8d270608c88b92c91ba47813a9f28d016

**Review by hoangmirs — APPROVED** [review]

File: .github/workflows/deploy_server_mock.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up flyctl
+        uses: superfly/flyctl-actions/setup-flyctl@master
+
+      - name: Deploy to fly
+        run: flyctl deploy --remote-only
+        env:
+          FLY_API_TOKEN: ${{ secrets.FLY_API_TOKEN }}
```
Comment (sanG-github) [inline .github/workflows/deploy_server_mock.yml:26]: Just a note, please make sure that we will notice others put this variable `FLY_API_TOKEN` into secrets. 😜

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,7 @@
+FROM stoplight/prism:latest
+
+COPY /docs /usr/src/prism/packages/cli/docs/
+
+EXPOSE 80
+
+CMD ["mock", "-h", "0.0.0.0", "-p", "80", "docs/openapi/openapi.yml"]
```
Comment (sanG-github) [inline Dockerfile:7]: Not sure if it should be the path from the Docker Image? 🤔 

```suggestion
CMD ["mock", "-h", "0.0.0.0", "-p", "80", "/usr/src/prism/packages/cli/docs/openapi/openapi.yml"]
```

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,10 @@
+.PHONY: install-dependencies doc/generate dev
+
+install-dependencies:
+	yarn install
```
Comment (sanG-github) [inline Makefile:4]: Please help to check the indent. ✨ (using tab with 2 spaces instead)
```suggestion
  yarn install
```

Comment (suho) [reply Makefile:4]: @sanG-github For Makefile, `TAB` is required

File: Dockerfile (hunk 3 above)
Comment (suho) [reply Dockerfile:7]: Yes, but as we copied to that path ([this line](https://github.com/nimblehq/api-mock-templates/pull/3/files/9cb6f6d8d270608c88b92c91ba47813a9f28d016#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R3)). So I think it's good to use the short path instead

Comment (sanG-github) [reply Dockerfile:7]: Ahh my bad. Thanks for your explanation.

File: Makefile (hunk 4 above)
Comment (sanG-github) [reply Makefile:4]: Ah okay, it shows `TAB with 2 spaces` in my local, but seems like it shows `TAB with 8 spaces` here 😢 

| Local                                                                                                                                   	| Remote                                                                                                                                  	|
|-----------------------------------------------------------------------------------------------------------------------------------------	|-----------------------------------------------------------------------------------------------------------------------------------------	|
| image 	| image 	|

**Review by sanG-github — APPROVED** [review]

**Pull request merged**